### DISCUSSION TITLE
SMARTS for ring membership in two different sized rings doesn't work

### DISCUSSION THREAD

**Discussion by zackterray** [rdkit/rdkit#6466] [Q&A]
I want to use a smarts pattern for a a bridgehead atom with membership in a 5 and 6 membered ring.  The obvious smarts for this would just be **'[r5;r6]'**, but this pattern does not match any of these examples:
**c1(CCC2)c2cccc1**
**c12cc[nH]c1cccc2**

What's even weirder is the two examples above match the pattern for atom in 5m ring with two ring membership: **'[r5;R2]'**
but they do not match the smarts for 6m with two ring membership: **'[r6;R2]'**

rdkit version = 2020.09.1

**Comment by bp-kelley** [comment]
The definition or 'r' and 'R' in daylight smarts is:  

```
r<n>  - in smallest SSSR ring of size <n>
R<n> - in <n> SSSR rings
```

The RDKit interprets r as "is the atom's smallest ring size equal to n"

From your first molecule this can be seen here:

```
>>> p = Chem.MolFromSmarts('[r5]')
>>> mol.GetSubstructMatches(p)
((0,), (1,), (2,), (3,), (4,))
>>> p = Chem.MolFromSmarts('[r6]')
>>> mol.GetSubstructMatches(p)
((5,), (6,), (7,), (8,))
```

So there is no overlap.  We can make a recursive smarts, however:

```
>>> p = Chem.MolFromSmarts('[$([r5&R2][r6])]')
>>> mol.GetSubstructMatches(p)
((0,), (4,))
```

This translates to, "match any atom in two rings, one of size 5 that is attached to an atom in a ring of size 6".  You can learn more about recursive smarts here:

https://www.daylight.com/dayhtml/doc/theory/theory.smarts.html

Hope this helps.
